### PR TITLE
Apply OkHttp listener auto finish timestamp to all running spans

### PR DESCRIPTION
**Pull request by stefanosiano**
… to http root call span

## :scroll: Description
scheduled finish timestamp now applies to all running spans, not only to http root call span

## :bulb: Motivation and Context
When the response is not read, we finish the OkHttp span automatically.
We were finishing the root call span at the last http span timestamp, but we were not applying it to all other OkHttp running spans.
[An image is worth more than a thousand words](https://demo.sentry.io/performance/android:c2475d976c3942c296351f558a7dc936/?limit=100&transaction=EmpowerPlantActivity)

## :green_heart: How did you test it?
Unit test

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt [hunk 1]
```diff
@@ -158,10 +157,12 @@ internal class SentryOkHttpEvent(private val hub: IHub, private val request: Req
 
         // We forcefully finish all spans, even if they should already have been finished through finishSpan()
         eventSpans.values.filter { !it.isFinished }.forEach {
-            // If a status was set on the span, we use that, otherwise we set its status as error.
-            it.status = it.status ?: SpanStatus.INTERNAL_ERROR
             moveThrowableToRootSpan(it)
-            it.finish()
+            if (finishDate != null) {
+                it.finish(it.status, finishDate)
```
Comment (markushi) [inline sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt:162]: Shouldn't this be
```suggestion
                it.finish(it.status ?: SpanStatus.INTERNAL_ERROR, finishDate)
```

**Review by markushi — CHANGES_REQUESTED** [review]
LGTM! I have one question I'd like to get answered before merging it. 😅

File: sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt (hunk 1 above)
Comment (stefanosiano) [reply sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt:162]: it was, but it actually didn't make sense, at least in the case linked in this PR.
If the response is not read, there will be some running spans, and they don't have any error.
So they didn't have to be reported as errored in the first place, i guess

Comment (stefanosiano) [reply sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt:162]: If there is another possible use case, we can think more about it

Comment (markushi) [reply sentry-okhttp/src/main/java/io/sentry/okhttp/SentryOkHttpEvent.kt:162]: 👍 makes a lot of sense this way then!

**Review by markushi — APPROVED** [review]

**Pull request merged**